OpenRA.Mods.Common/Traits/Render/WithBarrel.cs

```diff
@@ -51,38 +52,43 @@ public IEnumerable<IActorPreview> RenderPreviewSprites(ActorPreviewInitializer i
 		}
 	}
 
-	class WithBarrel
+	public class WithBarrel : UpgradableTrait<WithBarrelInfo>
 	{
-		WithBarrelInfo info;
-		Actor self;
-		Armament armament;
-		Turreted turreted;
-		IBodyOrientation body;
-		Animation anim;
+		public readonly Animation DefaultAnimation;
```

Comment:
This doesn't appear to be used anywhere, so why make it public?

If you need to e.g. run an animation from an external trait then that should be done via an appropriate interface.